templates/cicd/pipeline.yml

```diff
@@ -22,5 +22,5 @@ stages:{{range .Stages}}
     - # The name of the environment to deploy to.
       name: {{.Name}}
       # Optional: use test commands to validate this stage of your build.
-      # test_commands: {{.TestCommands}}
+      # test_commands: [echo 'running tests', make test]
```

Comment:
This is a change to the default, right? Is there an advantage to hardcoding these in the template file instead of the output of CreatePipeline?

Reply:
I think it lends discoverability of this option to this process. And as of right now, the pipeline.yml file is the only spot where users can input their test commands.

Reply:
The field is commented out because it's optional, but a couple examples are given as a guide.

Reply:
When `pipeline init` is run, NewPipeline() is called, which generates and returns a new Pipeline struct, which presumably can hold the .TestCommands field. 

I think that we should move the default commands to a const in the pipeline go code and use the `{{.TestCommands}}` syntax for the template since the templates are far more fragile and less lintable than go code, and we should minimize future changes which require us to write to them. This will let us update the default test commands easily with minimal risk of bugs if we should ever want to change them.

Reply:
Ahh, now I see what you mean. Sorry I misunderstood. 

The hope is to keep this as a nil value unless it's used, in which case it's instantiated in this step. So "[echo 'running tests', make test]" are not so much default commands as dummy placeholders that give users an example of the syntax. If line 25 is not uncommented (and likely replaced with the user's own commands), manifest.PipelineStage.TestCommands will still be nil, and after convertStages() in pipeline_update, deploy.PipelineStage.TestCommands will also be nil. I see how this diff makes it look like something else happened here!  

I agree with you that if these _were_ default values, they should be set when `pipeline init` is run.